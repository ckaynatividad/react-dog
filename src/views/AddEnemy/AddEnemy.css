.form2 {
  display: flex;
  justify-content: center;
  color: white;
  line-height: 40px;
}
