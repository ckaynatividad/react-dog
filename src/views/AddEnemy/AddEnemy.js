import React, { useState } from 'react';
import { useHistory } from 'react-router';
import AdminForm from '../../components/Admin/Admin';
import { createEnemy } from '../../services/Enemies';
import './AddEnemy.css';

export default function AddEnemy() {
  const [enemy, setEnemy] = useState({});

  const history = useHistory();
    
  const handleSubmit = async (e) => {
    e.preventDefault();
    try { await createEnemy(enemy) ;
      alert('Nice, added');
      history.push(`/`);
    } catch {
      alert('Try again');
    }
  };

  const setNew = (key, value) => {
    enemy[key] = value;
    setEnemy({ ...enemy });
  };
  return (
    <div className="form2">
      <AdminForm enemy={enemy} handleSubmit={handleSubmit} setNew={setNew}/>

    </div>
  );
}
