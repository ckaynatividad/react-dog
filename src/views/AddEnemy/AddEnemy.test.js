import { render, screen } from '@testing-library/react';
import AddEnemy from './AddEnemy';

test('add enemy form should; render', () => {
  const container = render(
    <AddEnemy />
  );
  expect(container).toMatchSnapshot();
});