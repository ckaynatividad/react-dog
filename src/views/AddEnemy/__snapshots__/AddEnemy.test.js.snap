// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add enemy form should; render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form2"
      >
        <div
          class="form"
        >
          <form>
            <label>
              name
            </label>
            <input
              placeholder="name"
              type="text"
              value=""
            />
            <label>
              level
            </label>
            <input
              placeholder="level"
              type="number"
              value=""
            />
            <label>
              type
            </label>
            <input
              placeholder="type"
              type="text"
              value=""
            />
            <label>
              image
            </label>
            <input
              placeholder="image url"
              type="text"
              value=""
            />
            <label>
              element
            </label>
            <input
              placeholder="element"
              type="text"
              value=""
            />
            <p>
              <button>
                submit
              </button>
            </p>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form2"
    >
      <div
        class="form"
      >
        <form>
          <label>
            name
          </label>
          <input
            placeholder="name"
            type="text"
            value=""
          />
          <label>
            level
          </label>
          <input
            placeholder="level"
            type="number"
            value=""
          />
          <label>
            type
          </label>
          <input
            placeholder="type"
            type="text"
            value=""
          />
          <label>
            image
          </label>
          <input
            placeholder="image url"
            type="text"
            value=""
          />
          <label>
            element
          </label>
          <input
            placeholder="element"
            type="text"
            value=""
          />
          <p>
            <button>
              submit
            </button>
          </p>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
